[Wayne Zhang Si Yuan] iP

waynezsy opened the pull request:
## PR Description Heading

This is a bullet list:

*  Item 1
*  Item 2
*  Item 3

This is a number list:

1.  Item 1
1.  Item 2
1.  Item 3

```java
public class Main {
    public void main(String[] args) {
        System.out.println("Hi I am a fenced code block with syntax highlighting.");
    }
}
```

This is a task list with emojis:

- [x] Completed Task 😄 
- [ ] Incomplete Task 😢 

> Am I a quote? - Confused quote

[Hyper link to my iP repo](https://github.com/waynezsy/ip)

Here be some `inline code`.

This sentence features text formatting such as **bold**, _italic_ and ~~strikethrough~~.

Commit:
Add Gradle support

waynezsy committed:
Duke class: implement skeletal version

Duke has no functionality.

Let's,
* add a greeting on start
* echo commands entered by the user
* add an exit command

waynezsy committed:
Duke class: change formatting of messages

Duke prints messages on successive lines.

Lines and indentations differentiate messages between users and Duke
more clearly.

Let's add lines and indentations to Duke's messages.

waynezsy committed:
Add ability to store text and display as needed

waynezsy committed:
Add ability to mark tasks as done (or not)

waynezsy committed:
Add support for tracking 3 types of tasks

waynezsy committed:
Add automated text UI testing

waynezsy committed:
Fix minor errors in javadocs

waynezsy committed:
Duke class: handles errors from incorrect inputs

waynezsy committed:
Duke class: store tasks with ArrayList

waynezsy committed:
Add support for deleting tasks

waynezsy committed:
Use Enum for Duke::execute

waynezsy committed:
Fix minor style errors

waynezsy committed:
Add ability to save tasks into hard disk

waynezsy committed:
Fix minor style errors

waynezsy committed:
Add ability to parse dates

waynezsy committed:
Merge branch 'branch-Level-7'

waynezsy committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/Task.java
#	src/main/java/Todo.java

waynezsy committed:
Refactor Duke class into multiple classes

waynezsy committed:
Organise Java classes into packages

waynezsy committed:
Remove data/tasks.txt from revision control

waynezsy committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

waynezsy committed:
Set up Gradle

waynezsy committed:
Add JUnit tests for Parser, Task and TaskList

waynezsy committed:
Add JavaDoc for class fields and fix minor errors

waynezsy committed:
Tweak code to comply with coding standard

waynezsy committed:
Add ability to find tasks by searching a keyword

waynezsy committed:
Merge branch 'branch-A-JavaDoc'

waynezsy committed:
Merge branch 'branch-A-CodingStandard'

waynezsy committed:
Merge branch 'branch-Level-9'

waynezsy committed:
Parser class: fix coding standard violations

waynezsy committed:
Merge branch 'branch-Level-9'

waynezsy committed:
TaskList class: use varargs for constructor

waynezsy committed:
Update build.gradle for JavaFX

Hikoya commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,176 @@
+ * The Ui class deals with interactions with the user.
+ */
+public class Ui {
+    private static final String INDENTATION = "    ";
```
Perhaps you can use a common prefix for all 3 constants as they serve similar functions? (eg. printing out to console)

Hikoya commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,36 @@
+    }
+
+    /**
+     * {@inheritDoc}
```
Maybe you should add a few more comments on what is the different behavior of this overridden function? eg. saving to File etc. I noticed the same issue in several other places too.

waynezsy replied:
Admittedly got a bit lazy, will probably update the JavaDoc comments for the execute() methods in each subclass to better clarify what they do in a future commit :) Thanks!

Hikoya commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,129 @@
+    /**
+     * Marks a Task as not done.
+     */
+    public void unmarkAsDone() {
```
```Suggestion
    public void markAsUndone() {
```

Hikoya reviewed (commented):
LGTM, coding style adheres to the standard very well, except for a few nits here and there. Keep up the good work!

waynezsy committed:
Implement GUI for Duke

waynezsy committed:
Update JavaDoc for Ui and MainWindow classes

waynezsy committed:
Merge branch 'branch-Level-10'

waynezsy committed:
Merge branch 'branch-A-Varargs'

waynezsy committed:
Add colours to GUI

waynezsy committed:
Remove text-ui-test

waynezsy committed:
Update GUI colour palette

lyuting47 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,143 @@
+                    throw new DukeException("Specify which task to delete with a single integer.");
+                }
+
+            case ON:
+                if (fullCommand.length() == 2) {
+                    throw new DukeException("Specify the date to check with yyyy-MM-dd.");
+                }
+                try {
+                    LocalDate date = LocalDate.parse(fullCommand.substring(3));
+                    return new OnCommand(date);
+                } catch (DateTimeParseException e) {
+                    throw new DukeException("Specify the date to check with yyyy-MM-dd.");
+                }
```
Nice additional feature that makes good use of the Datetime parser!

lyuting47 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,129 @@
+     * @param date The specified date to check.
+     * @return true if the task occurs by/at the specified date.
+     */
+    public abstract boolean onDate(LocalDate date);
```
Maybe naming it as isOnDate makes it clearer that this method returns a boolean?

lyuting47 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,192 @@
+     * @param keyword The specified keyword (or phrase) within the descriptions of tasks.
+     * @return A new TaskList of every task which description contains a specified keyword (or phrase).
+     */
+    public TaskList allContaining(String keyword) {
```
Changing this method name to getAllTasksContaining may make it more obvious what the method does.

lyuting47 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,192 @@
+     * @param date The specified date by/at which the tasks to find occur.
+     * @return A new TaskList of every task that occurs by/at a specified date.
+     */
+    public TaskList allOnDate(LocalDate date) {
```
Similarly for this, perhaps naming it getAllTasksOnDate may be clearer.

lyuting47 reviewed (commented):
Looks good! The code is very readable and follows OO principles closely. No major violations of the given coding standard either. Only some minor naming changes suggested.

waynezsy committed:
Enable GUI to display longer messages in full

When Duke's message is too long, the text gets shortened with ellipsis.

Users are unable to access their full task list when there are too many
items, limiting the usefulness of Duke's task list system.

Let's adjust height limitations for the Label and HBox in the DialogBox
to follow the preferred height needed to fit the whole message.

waynezsy committed:
Use Java assertions

There are important assumptions at various points in the code.

The use of assertions can verify if these assumptions hold true and
can help detect potential bugs.

Let's add assertions in some of the methods where the argument can be
null but should not be invoked with a null argument at any point.

waynezsy committed:
Improve overall code quality

The code base is written to fulfill functional requirements.

There are areas for improvement in terms of readability
and usage of unsafe practices.

Let's refactor the code to improve the code quality where necessary.

Improved code quality make it easier for future developers to continue
the project.

waynezsy committed:
TaskList class: use Streams where applicable

waynezsy committed:
Merge pull request #1 from waynezsy/branch-A-Assertions

Use Java assertions

waynezsy committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/TaskList.java
#	src/main/java/duke/task/Todo.java

waynezsy committed:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	src/main/java/duke/task/TaskList.java

waynezsy committed:
Merge pull request #2 from waynezsy/branch-A-CodeQuality

Improve overall code quality

waynezsy committed:
Merge branch 'master' into branch-A-Streams

# Conflicts:
#	src/main/java/duke/task/TaskList.java

waynezsy committed:
Merge pull request #3 from waynezsy/branch-A-Streams

TaskList class: use Streams where applicable

waynezsy committed:
Set up Continuous Integration

waynezsy committed:
Fix Storage::writeToFile bug

The writeToFile method is calling a renamed method by its old name.

The program is unable to run in its current state due to this grave bug.

Let's update the body of writeToFile to invoke
TaskList::convertAllToString by its new name to fix the program.

waynezsy committed:
Enable operating on multiple tasks

Duke can only delete, mark or unmark a single task at a time.

It is more convenient to delete, mark or unmark multiple tasks with a
single user input.

Let's enable delete, mark and unmark commands to detect multiple
integers and operate on all the specified tasks.

waynezsy committed:
Fix bugs in TaskListTest class

waynezsy committed:
Merge pull request #4 from waynezsy/branch-C-MassOps

Enable operating on multiple tasks

waynezsy committed:
Refactor Duke class to NyanDuke class

waynezsy committed:
Fix MassOps bug

waynezsy committed:
Add product screenshot

waynezsy committed:
Add user guide for NyanDuke to docs/README.md

waynezsy committed:
Fix user guide format on product site

waynezsy committed:
Fix minor user guide format bug

waynezsy committed:
Improve readability of user guide

waynezsy committed:
Fix edge case bugs

waynezsy committed:
Enable NyanDuke to detect empty strings

waynezsy committed:
Fix user guide mistakes under Usage

waynezsy committed:
Update Javadoc comments

waynezsy committed:
Optimise GUI and exit behaviour

waynezsy committed:
Refactor error messages to constants

waynezsy committed:
Update build.gradle for Linux
